src/lua/database.c

```diff
@@ -206,6 +206,32 @@ static int database_numindex(lua_State *L)
   return 1;
 }
 
+static int database_get_image(lua_State *L)
+{
+  const int img_id = luaL_checkinteger(L, -1);
+  if(img_id < 1)
+  {
+    return luaL_error(L, "incorrect image id in database");
+  }
+  sqlite3_stmt *stmt = NULL;
+  char query[1024];
+  snprintf(query, sizeof(query), "SELECT id FROM main.images WHERE id = %d LIMIT 1",
+           img_id);
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), query, -1, &stmt, NULL);
+  if(sqlite3_step(stmt) == SQLITE_ROW)
+  {
+    int imgid = sqlite3_column_int(stmt, 0);
+    luaA_push(L, dt_lua_image_t, &imgid);
+    sqlite3_finalize(stmt);
+  }
+  else
+  {
+    sqlite3_finalize(stmt);
+    lua_pushnil(L);
+  }
+  return 1;
```

Comment:
Why not a single `sqlite3_finalize(stmt);` after the if/else ?

Reply:
because I copied from above and just changed enough to make it work, instead of proof reading :).  I'll fix them both